src/agent/otubus.cpp

```diff
@@ -322,29 +326,26 @@ int UbusServer::UbusScanHandlerDetail(struct ubus_context *     aContext,
 
     otError  error = OT_ERROR_NONE;
     uint64_t eventNum;
-    ssize_t  retval;
 
-    blob_buf_init(&mBuf, 0);
-    sJsonUri = blobmsg_open_array(&mBuf, "scan_list");
+    memset(&mScanReplyBuf, 0, sizeof(mScanReplyBuf));
+    blob_buf_init(&mScanReplyBuf, 0);
+    sJsonUri = blobmsg_open_array(&mScanReplyBuf, "scan_list");
 
     mIfFinishScan = 0;
     sUbusServerInstance->ProcessScan();
 
+    // XXX: Is there a neccessity to manually trigger openthread event loop?
```

Comment:
If not trigger, the openthread event loop won't run.. (But this is the previous situation, maybe now is OK, needs some experiments).